asmcli/lib/validate.sh

```diff
@@ -250,6 +251,26 @@ validate_ca_consistency() {
   fi
 }
 
+validate_no_ingress_gateway() {
+  local INGRESS_GATEWAY
+  INGRESS_GATEWAY="$(kubectl get deployments -n istio-system -l operator.istio.io/component=IngressGateways)"
+  if [[ -z "${INGRESS_GATEWAY}" ]]; then
+    return
+  fi
+
+  local CUSTOM_OVERLAY; CUSTOM_OVERLAY="$(context_get-option "CUSTOM_OVERLAY")"
+  if [[ "${CUSTOM_OVERLAY}" =~ "legacy-default-ingressgateway" ]]; then
+    return
+  fi
```

Comment:
This check is cheaper so it should probably go first.

Reply:
Done